R/zzz.R

```diff
@@ -18,17 +18,17 @@ gradethis_default_options <- list(
 
 
   ### legacy ###
-  gradethis_glue_correct = "{ random_praise() } { .message } { .correct }",
-  gradethis_glue_incorrect = "{ .message } { .incorrect } { random_encouragement() }",
+  gradethis_glue_correct = "{random_praise()} {.message} {.correct}",
+  gradethis_glue_incorrect = "{pipe_warning()}{.message} {.incorrect} {random_encouragement()}",
 
-  gradethis_glue_pipe = paste0(
+  gradethis.pipe_warning = paste0(
     "I see that you are using pipe operators (e.g. %>%), ",
     "so I want to let you know that this is how I am interpretting your code ",
-    "before I check it:\n\n```r\n{unpipe_all_str(.user, width = 60)}\n```\n\n{.message}"
+    "before I check it:\n\n```r\n{.user_code_unpiped}\n```\n\n"
   ),
```

Comment:
Now that this is not _legacy_, should we move it up the the upper group?

Reply:
This one would be the exception to that rule. Technically, `pipe_warning()` creates the message but it assembles `.user_code_unpiped` inside the function. I don't think we want to export `unpipe_all_str()`, so I think we're stuck with it.

I guess I'd amend the rule to say that any helpers used in the default arguments of top-level functions should also be functions. So in this case `pipe_warning()` is called at the top level in the glue-string of `gradethis.code.incorrect`, but `gradethis.pipe_warning` is the default value of `message` in `pipe_warning()`, or two levels deep, so it's okay if it uses a non-function value in the glue string.

Reply:
My fault for not being clear.

Should we move the `gradethis.pipe_warning` definition above line 20 as `gradethis.pipe_warning` is not just for soft deprecated functions?

Reply:
oh I see, yeah that makes sense